okta/komand_okta/triggers/users_added_removed_from_group/trigger.py

```diff
@@ -0,0 +1,108 @@
+                                      assistance="Double-check that group ID's are all valid.",
+                                      data=response.text)
+            if response.status_code not in range(200, 299):
+                error_code = data["errorCode"]
+                error_summary = data["errorSummary"]
+                error_causes = data["errorCauses"]
+                raise PluginException(cause=f"Okta returned a {response.status_code} status code, and a error code of {error_code}.",
+                                      assistance=f"Summary: {error_summary}. Possible causes: {error_causes}.",
+                                      data=response.text)
+            data = komand.helper.clean(data)
+            current_list.append({group: data})
+
+            # Get group names
+            group_name_api = f"{okta_url}/api/v1/groups/{group}"
+            response = self.connection.session.get(group_name_api)
+            try:
+                data = response.json()
+            except ValueError:
+                raise PluginException(cause='Returned data was not in JSON format.',
+                                      assistance="Double check that group ID's are all valid.",
+                                      data=response.text)
+            if response.status_code not in range(200, 299):
+                error_code = data["errorCode"]
+                error_summary = data["errorSummary"]
+                error_causes = data["errorCauses"]
+                raise PluginException(cause=f"Okta returned a {response.status_code} status code, and a error code of {error_code}.",
+                                      assistance=f"Summary: {error_summary}. Possible causes: {error_causes}.",
+                                      data=response.text)
+            group_names.append(data["profile"]["name"])
+
+        while True:
+            new_list = list()
+            for group in group_list:
+                api = f"{okta_url}/api/v1/groups/{group}/users"
+
+                response = self.connection.session.get(api)
+
+                try:
+                    data = response.json()
+                except ValueError:
+                    raise PluginException(cause='Returned data was not in JSON format.',
+                                          assistance="Double check that group ID's are all valid.",
+                                          data=response.text)
+                if response.status_code not in range(200, 299):
+                    error_code = data["errorCode"]
+                    error_summary = data["errorSummary"]
+                    error_causes = data["errorCauses"]
+                    raise PluginException(
+                        cause=f"Okta returned a {response.status_code} status code, and a error code of {error_code}.",
+                        assistance=f"Summary: {error_summary}. Possible causes: {error_causes}.",
+                        data=response.text)
```

Comment:
Any way that we could move this into a function? Maybe one that takes a response and then raises a PluginException based on the error? We could reuse that across the plugin, then.